### ISSUE TITLE
Do not await in assertion, await and then assert instead

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#2173]
We should report an issue in case something is awaited within an assertion and propose to await and then assert. That makes the code easier to read and follow.

Following should cause a warning.
```c#
Assert.That(await xyz.DoSomething(), Is.True);
```

It should be fixed into
```c#
var awaitedResult = await xyz.DoSomething();

Assert.That(awaitedResult, Is.True);
```
